jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Attributes.java

```diff
@@ -135,6 +136,34 @@ static Attributes unwrap(Attributes attributes)
         return attributes;
     }
 
+    static int hashCode(Attributes attributes)
+    {
+        int hash = 113;
+        for (String name : attributes.getAttributeNameSet())
+            hash += name.hashCode() ^ attributes.getAttribute(name).hashCode();
+        return hash;
+    }
```

Comment:
I don't think we want to be fancy and deviate from widely-known hash functions.

```suggestion
    static int hashCode(Attributes attributes)
    {
        int hash = 0;
        for (String name : attributes.getAttributeNameSet())
        {
            hash = 31 * hash + Objects.hash(name, attributes.get(name));
        }
        return hash;
    }
```
FTR, `31` seems not a good number either (https://stackoverflow.com/questions/1835976/what-is-a-sensible-prime-for-hashcode-calculation/2816747#2816747), but it's what Java uses and the algorithm is right.